src/harness/fake_sas.py

```diff
@@ -409,16 +409,40 @@ def do_GET(self):
     self.wfile.write(json.dumps(response))
 
 
-def RunFakeServer(version, is_ecc):
+def RunFakeServer(version, is_ecc, ca_cert, verify_crl):
   FakeSasHandler.SetVersion(version)
   if is_ecc:
     assert ssl.HAS_ECDH
+  if ca_cert is not None:
+    try:
+      with open(os.path.join('certs', ca_cert)) as file_handle:
+        ca_cert_data = file_handle.read()
```

Comment:
Instead of discarding the contents, does `ssl_context.load_verify_locations(cadata=ca_cert_data)` work?

Reply:
Looks   like the cadata argument does not accept CRLs. There seems to be a bunch of   errors in the documentation for this method. First of all this argument   cadata does not work with ASCII characters and the input needs to be   converted to unicode. Second the statement "at least one of cafile or   capath must be specified." is not correct. Third and most important is   that cadata doesn't work when we appended the certificate chain with   CRLs.     The following are the bugs being tracked.     https://bugs.python.org/issue29236     https://bugs.python.org/issue22365          Hence we have updated the code to use cafile parameter with the   ca_cert_path parameter determined above.